### PR TITLE
Make controller base_image version an argument

### PR DESCRIPTION
**Pull request by michaelhtm**
Description of changes:
This change allows us to pass the base image version 
as an argument allowing us to pass the latest one after each release

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/build-controller-image.sh [hunk 1]
```diff
@@ -88,9 +89,17 @@ if ! is_public_ecr_logged_in; then
...
+  GOLANG_VERSION=$(cat build_config.yaml | yq .go_version)
+  BASE_IMAGE_VERSION=$(cat build_config.yaml | yq .eks_distro_version) 
+popd 1>/dev/null
+
+
 pushd "$ROOT_DIR" 1>/dev/null
   # Get the golang version from the code-generator
   GOLANG_VERSION=${GOLANG_VERSION:-"$(go list -f {{.GoVersion}} -m)"}
+  BASE_IMAGE_VERSION=${BASE_IMAGE_VERSION:-"$(echo 2024-08-13-1723575672.2)"}
 popd 1>/dev/null
```
Comment (a-hilaly) [inline scripts/build-controller-image.sh:103]: We're doing this twice should we get rid of the ROOT_DIR one?

File: scripts/build-controller-image.sh [hunk 2]
```diff
@@ -88,9 +89,17 @@ if ! is_public_ecr_logged_in; then
...
+  # Get the golang version from build_config.yaml
+  GOLANG_VERSION=$(cat build_config.yaml | yq .go_version)
+  BASE_IMAGE_VERSION=$(cat build_config.yaml | yq .eks_distro_version) 
+popd 1>/dev/null
+
+
 pushd "$ROOT_DIR" 1>/dev/null
   # Get the golang version from the code-generator
   GOLANG_VERSION=${GOLANG_VERSION:-"$(go list -f {{.GoVersion}} -m)"}
+  BASE_IMAGE_VERSION=${BASE_IMAGE_VERSION:-"$(echo 2024-08-13-1723575672.2)"}
```
Comment (a-hilaly) [inline scripts/build-controller-image.sh:102]: I think we should fail if we're missing a GOLANG_VERSION/BASE_IMAGE_VERSION

File: scripts/build-controller-image.sh (hunk 1 above)
Comment (michaelhtm) [reply scripts/build-controller-image.sh:103]: this is a failsafe, just in case the image is not being built with test-infra in the expected path

**Review by a-hilaly — APPROVED** [review]
👍 
/lgtm

**Pull request merged**